### ISSUE TITLE
WP "user actions" breaks waypoint navigation

### ISSUE THREAD

**Issue by ABLomas** [iNavFlight/inav-configurator#1721]
Short video demonstrating issue:
https://user-images.githubusercontent.com/8989151/218552591-65620b35-67e2-451c-ae52-334931ab7444.mp4
Text version - mission waypoint with user action set has it's altitude set to MSL+preset altitude, while AGL setting is selected.
This was registered as issue on another branch but it looks like firmware has nothing to do with it:
https://github.com/iNavFlight/inav/issues/8798

In short - looks like INAV-configurator incorrectly parses bit0 in P3 field and and this is reason why alt is set to wrong value.

**Comment by breadoven** [comment]
Configurator needs updating to read the WP P3 parameter as a Bitfield rather than a boolean as it does at the moment.

**Comment by MrD-RC** [comment]
Configurator has been updated to read P3 as a bitfield. I thought I had tested that everything was toggled correctly. But something must have slipped through.

**Comment by breadoven** [comment]
OK missed that change. I was going to do it but seems you've saved me the trouble.

So just updated Configurator and checked this and it appears that the Bit is being set but not unset, so once it's been set it gets stuck true.

**Comment by breadoven** [comment]
The other problem is P3 is used as a boolean in various places which affects the altitude calculation depending on whether SL or relative altitude datum is used. Bit 0 needs checking rather than just the boolean state of the P3 value. I've made some changes that have fixed the obvious things, need to check other stuff isn't affected.

**Comment by breadoven** [comment]
Do you want me to fix this @MrD-RC given it's related to the MSL Altitude stuff I did originally ?

**Comment by MrD-RC** [comment]
@breadoven I found somewhere that was using P3 as a Boolean. So I've sorted that and fixed some other things too. The _Elevation (m)_ and _Grd Dist (m)_ seem to be working correctly to me. But, the elevation profile doesn't seem right. I couldn't see any references to P3 in there. Your thoughts? It may be fine though. It looks the same in 5.1 Configurator.

WP1
![image](https://user-images.githubusercontent.com/17590174/218828995-ccbedc3c-bae1-4157-bc99-dd767f6ff14e.png)

WP2
![image](https://user-images.githubusercontent.com/17590174/218829017-68d179a1-816e-43a8-ad8c-08cd56d8080e.png)

WP3
![image](https://user-images.githubusercontent.com/17590174/218829056-906caa8f-7d37-431a-a155-39907179ef5a.png)

WP4
![image](https://user-images.githubusercontent.com/17590174/218829081-b2321e71-d028-4244-aeef-885d90231fd4.png)

WP5
![image](https://user-images.githubusercontent.com/17590174/218829124-3b5c220a-4f13-4e0d-b9b5-d05781a9992d.png)

WP6
![image](https://user-images.githubusercontent.com/17590174/218829169-bfa9f6a9-b064-467c-8a94-43a2bad85811.png)

[test mission.mission.txt](https://github.com/iNavFlight/inav-configurator/files/10735841/test.mission.mission.txt)

**Comment by MrD-RC** [comment]
Added a PR https://github.com/iNavFlight/inav-configurator/pull/1722

**Comment by MrD-RC** [comment]
There is something causing issues in the elevation profile. If I disable the user actions, it looks as I would expect.

**Comment by MrD-RC** [comment]
Got it. It was hidden in `WaypointCollection`.

WP2 now for comparison
![image](https://user-images.githubusercontent.com/17590174/218835209-7283ae53-3b28-411d-a77c-ae5c865c293b.png)
